### PR TITLE
Fix bug#102, typor#103

### LINKED ISSUE #102: 3.3 中 原始字面量不一定是右值
## 实际描述

- 文件路径：例如，book/zh-cn/03-runtime.md
- 原文段落：

```
运算表达式产生的临时变量、 原始字面量、Lambda 表达式都属于纯右值。
```

## 原文表意不准确：

原始字面量并不一定是右值，比如 字符字面量，字符字面量可以被取地址，是一个左值。

参见：https://en.cppreference.com/w/cpp/language/value_category

### LINKED ISSUE #103: 笔误：3.3节：“需要拿到一个将亡值，就需要用到右值引用的申明：”
## 实际描述

- 文件路径：例如，book/zh-cn/03-runtime.md
- 原文段落：

```
需要拿到一个将亡值，就需要用到右值引用的申明：
```

## 预期描述

```
要拿到一个将亡值，就需要用到右值引用的声明：
```

### PR DESCRIPTION
**Pull request by cathaysia**
resolve #102 
resolve #103

## Description

fixed bug#102

## Change List

- Fix bug#102 of string literal is a rvalue just in class context

- Resolve error content of  book/en-US/03-runtime.md
---

## 说明

解决 bug#102, typor#103

## 变化箱单

- 修复了关于字符串仅在类中为右值的说明

### CODE REVIEW

File: book/zh-cn/03-runtime.md [hunk 1]
```diff
@@ -225,6 +225,22 @@ int main() {
 要么是求值结果相当于字面量或匿名临时对象，例如 `1+2`。非引用返回的临时变量、运算表达式产生的临时变量、
 原始字面量、Lambda 表达式都属于纯右值。
 
+需要注意的是：字符串字面量只有在类中才是右值，当其位于普通函数中是左值。例如：
+
+```cpp
+class Context{
```
Comment (changkun) [inline book/zh-cn/03-runtime.md:231]:
```suggestion
class Foo {
```

File: book/zh-cn/03-runtime.md [hunk 2]
```diff
@@ -225,6 +225,22 @@ int main() {
...
+        const char*&& right = "Hello, World"; // 此处字符串字面量为右值
+public:
+        void test(){
```
Comment (changkun) [inline book/zh-cn/03-runtime.md:234]:
```suggestion
        void bar(){
```

File: book/zh-cn/03-runtime.md [hunk 3]
```diff
@@ -225,6 +225,22 @@ int main() {
...
+            right = "Also Right"; // 此处字符串字面量为右值
```
Comment (changkun) [inline book/zh-cn/03-runtime.md:235]:
```suggestion
            right = "still rvalue"; // 此处字符串字面量为右值
```

File: book/zh-cn/03-runtime.md [hunk 4]
```diff
@@ -225,6 +225,22 @@ int main() {
 要么是求值结果相当于字面量或匿名临时对象，例如 `1+2`。非引用返回的临时变量、运算表达式产生的临时变量、
 原始字面量、Lambda 表达式都属于纯右值。
 
+需要注意的是：字符串字面量只有在类中才是右值，当其位于普通函数中是左值。例如：
+
+```cpp
+class Context{
+        const char*&& right = "Hello, World"; // 此处字符串字面量为右值
```
Comment (changkun) [inline book/zh-cn/03-runtime.md:232]:
```suggestion
        const char*&& right = "this is a rvalue"; // 此处字符串字面量为右值
```

File: book/zh-cn/03-runtime.md [hunk 5]
```diff
@@ -225,6 +225,22 @@ int main() {
...
+public:
+        void test(){
+            right = "Also Right"; // 此处字符串字面量为右值
+        } 
+};
+
+int main(){
+    const char* const &left = "this is Left"; // 此处字符串字面量为左值
```
Comment (changkun) [inline book/zh-cn/03-runtime.md:240]:
```suggestion
    const char* const &left = "this is an lvalue"; // 此处字符串字面量为左值
```

**Comment by cathaysia** [comment]
All the suggestions should bave been adapted at the last commit.

File: book/en-us/03-runtime.md [hunk 6]
```diff
@@ -260,6 +260,22 @@ Temporary variables returned by non-references, temporary variables generated
...
+class Context{
+        const char*&& right = "Hello, World"; // here string literal is a rvalue
+public:
+        void test(){
+            right = "Also Right"; // here string literal is a rvalue
+        } 
+};
+
+int main(){
+    const char* const &left = "this is Left"; // here string literal is a lvalue
```
[the remark this answers is not in the document]
Comment (changkun) [reply book/en-us/03-runtime.md:275]: Please fix here

**Comment by changkun** [comment]
Sorry, but you have one more left

File: book/en-us/03-runtime.md [hunk 7]
```diff
@@ -260,6 +260,22 @@ Temporary variables returned by non-references, temporary variables generated
 by operation expressions, original literals, and Lambda expressions 
 are all pure rvalue values.
 
+Note that a string literal became rvalue in a class, and remains an lvalue in other cases (e.g., in a function)：
+
+```cpp
+class Foo {
+        const char*&& right = "this is a rvalue";
+public:
+        void bar(){
```
Comment (changkun) [inline book/en-us/03-runtime.md:269]:
```suggestion
        void bar() {
```

File: book/en-us/03-runtime.md [hunk 8]
```diff
@@ -260,6 +260,22 @@ Temporary variables returned by non-references, temporary variables generated
...
+            right = "still rvalue"; // the string literal is a rvalue
+        } 
+};
+
+int main(){
```
Comment (changkun) [inline book/en-us/03-runtime.md:274]:
```suggestion
int main() {
```

File: book/en-us/03-runtime.md [hunk 9]
```diff
@@ -260,6 +260,22 @@ Temporary variables returned by non-references, temporary variables generated
...
 
+Note that a string literal became rvalue in a class, and remains an lvalue in other cases (e.g., in a function)：
+
+```cpp
+class Foo {
+        const char*&& right = "this is a rvalue";
+public:
+        void bar(){
+            right = "still rvalue"; // the string literal is a rvalue
+        }
```
Comment (changkun) [inline book/en-us/03-runtime.md:271]:
```suggestion
        }
```

File: book/zh-cn/03-runtime.md [hunk 10]
```diff
@@ -225,6 +225,22 @@ int main() {
 要么是求值结果相当于字面量或匿名临时对象，例如 `1+2`。非引用返回的临时变量、运算表达式产生的临时变量、
 原始字面量、Lambda 表达式都属于纯右值。
 
+需要注意的是，字符串字面量只有在类中才是右值，当其位于普通函数中是左值。例如：
+
+```cpp
+class Foo{
```
Comment (changkun) [inline book/zh-cn/03-runtime.md:231]:
```suggestion
class Foo {
```

File: book/zh-cn/03-runtime.md [hunk 11]
```diff
@@ -225,6 +225,22 @@ int main() {
...
+        const char*&& right = "this is a rvalue"; // 此处字符串字面量为右值
+public:
+        void bar(){
```
Comment (changkun) [inline book/zh-cn/03-runtime.md:234]:
```suggestion
        void bar() {
```

File: book/zh-cn/03-runtime.md [hunk 12]
```diff
@@ -225,6 +225,22 @@ int main() {
...
+            right = "still rvalue"; // 此处字符串字面量为右值
+        } 
+};
+
+int main(){
```
Comment (changkun) [inline book/zh-cn/03-runtime.md:239]:
```suggestion
int main() {
```

**Comment by cathaysia** [comment]
Well, it should be ok now.

File: book/en-us/03-runtime.md [hunk 13]
```diff
@@ -260,6 +260,22 @@ Temporary variables returned by non-references, temporary variables generated
...
+class Foo {
+        const char*&& right = "this is a rvalue";
+public:
+        void bar() {
+            right = "still rvalue"; // the string literal is a rvalue
+        }
+};
+
+int main() {
+    const char* const &left = "this is Left"; // here string literal is a lvalue
```
Comment (changkun) [inline book/en-us/03-runtime.md:275]:
```suggestion
    const char* const &left = "this is an lvalue"; // the string literal is an lvalue
```

**Comment by cathaysia** [comment]
GitHub clone was too slow, work with webpage always make some troublesome . . . this time should be ok??

**Comment by changkun** [comment]
Thanks!

**Pull request merged**